MaxText/decode.py

```diff
@@ -220,12 +228,10 @@ def decode_loop(config, state=None):
   )
 
   new_cache = prefill_cache
-  outputs = []
   first_profiling_step = config.max_prefill_predict_length + config.skip_first_n_steps_for_profiler
   last_profiling_step = np.clip(first_profiling_step + config.profiler_steps - 1,
                                 first_profiling_step, config.max_target_length - 1)
 
-  #add the new_id which is the first generated token to outputs
   outputs = []
```

Comment:
we need to add the sample from new_logit in  L212 here

Reply:
Once I added code below

```
  # add the first generated token to outputs
  with jax.spmd_mode('allow_all'):
    first_token = inference_utils.sampling(next_logit, rng, config.decode_sampling_strategy,
                                          topk=config.decode_sampling_top_k, nucleus_topp=config.decode_sampling_nucleus_p,
                                          temperature=config.decode_sampling_temperature)
  outputs=[first_token]
```

It gave me `That That's great to hear! I love to learn new things and explore different interests` with extra `That`. Do you think this is expected? This will also change our matched results for Mistral model, right?